### ISSUE TITLE
test_instrument changes

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#182]
A few methods in test_instrument modify the loaded instrument module to ensure that pysat errors appropriately. These methods are currently fixed to pysat_testing, rather than the actual module under test. The first class within this file is subclassed by other testing classes, which causes the issue. See pull #179.

**Comment by jklenzing** [comment]
Should we combine the testing classes into a single one with different flags for output (pandas, xarray, etc)?  This might be a solution here, and it would reduce code duplication.

**Comment by rstoneback** [comment]
I’m not looking at the test routines this second but it certainly sounds reasonable to me.

On Jun 4, 2019, at 10:10 AM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


Should we combine the testing classes into a single one with different flags for output (pandas, xarray, etc)? This might be a solution here, and it would reduce code duplication.

**Comment by jklenzing** [comment]
`tag` and `sat_id` both used already in pysat_testing.  Keeping separate files for now, and working on this issue.

**Comment by jklenzing** [comment]
If we pass the `pandas_format` keyword through, we can switch between pandas and xarray for a single instrument.  I am trying this through the `init` routine in pysat_testing, however, I'm running into an error in `_instrument` where it checks for data type.

```
~/code/pysat/pysat/_instrument.py in _load_data(self, date, fid)
   1048         if not isinstance(data, self._data_library):
   1049             raise TypeError(' '.join(('Data returned by instrument load',
-> 1050                             'routine must be a', self._data_library)))
   1051         if not isinstance(mdata, _meta.Meta):
   1052             raise TypeError('Metadata returned must be a pysat.Meta object')
```

The issue seems to be that when I initialize the object, `_.instrument.__init__` has already set the `_data_library` as an empty `DataFrame`back in Lines 258-260 (see below).
```
        if self.pandas_format:
            self._null_data = DataFrame(None)
            self._data_library = DataFrame
        else:
            self._null_data = xr.Dataset(None)
            self._data_library = xr.Dataset
```

@rstoneback, any suggestions?

**Comment by jklenzing** [comment]
See https://github.com/pysat/pysat/blob/test_inst/pysat/instruments/pysat_testing.py#L130 for beginning of updates

**Comment by jklenzing** [comment]
There are some persistent issues with pysat.Instrument instantiating empty DataFrames when trying to pass an xarray option through pysat_testing `init`.  Consolidation of testing routines is beyond the scope of this issue.

**Comment by jklenzing** [comment]
I think that sounds like a good idea. I’ve written up the problem as part of #301 since a lot of this was me already adding to the scope of this issue.

**Comment by jklenzing** [comment]
Closed with merge of #298.
